### PR TITLE
Use unique_ptr for Driver::root_

### PR DESCRIPTION
**Pull request by oazizi000**
Change to use `std::unique_ptr` instead of raw pointer for `Driver::root_`, for the sake of more modern memory management.

##### Checklist

- [x] Language changes are updated in `man/adoc/bpftrace.adoc` and if needed in `docs/reference_guide.md`
- [x] User-visible and non-trivial changes updated in `CHANGELOG.md`
- [x] The new behaviour is covered by tests

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -397,16 +397,16 @@ static std::optional<struct timespec> get_boottime()
...
   err = driver.parse();
   if (err)
     return nullptr;
 
-  auto ast = driver.root_;
-  driver.root_ = nullptr;
-  return std::unique_ptr<ast::Node>(ast);
+  std::unique_ptr<ast::Node> retval = std::move(driver.root_);
+  driver.root_.reset();
+  return retval;
```
Comment (thefallentree) [inline src/main.cpp:409]: I feel this could be just `return (ast::Node *)driver.root_.release();`  ?

**Review by thefallentree — APPROVED** [review]

**Review by thefallentree — CHANGES_REQUESTED** [review]
pls fix CI, looks like it needs a cast

File: src/main.cpp (hunk 1 above)
Comment (oazizi000) [reply src/main.cpp:409]: Yup, this can be simplified. Changing it to `return std::move(driver.root_)`.

**Review by thefallentree — APPROVED** [review]

File: src/driver.cpp [hunk 2]
```diff
@@ -19,7 +19,6 @@ Driver::Driver(BPFtrace &bpftrace, std::ostream &o)
 
 Driver::~Driver()
 {
-  delete root_;
 }
```
Comment (danobi) [inline src/driver.cpp:22]: Just delete the destructor entirely?

**Comment by fbs** [comment]
Looks good.

To be nitpicky, we should rename root_ to root. We try to use _ for private members

File: src/driver.cpp (hunk 2 above)
Comment (oazizi000) [reply src/driver.cpp]: Indeed...that's cleaner.

**Comment by oazizi000** [comment]
> Looks good.
> 
> To be nitpicky, we should rename root_ to root. We try to use _ for private members



> Looks good.
> 
> To be nitpicky, we should rename root_ to root. We try to use _ for private members

Went ahead and changed `root_` to `root`.

**Pull request merged**